speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/DecisionMaker.inl

```diff
@@ -5,32 +5,35 @@
 
 /// @brief  Creates an implementation of a decision maker
 #define CREATE_DECISION_MAKER_IMPLEMENTATION(type1,type2) \
-    template void DecisionMaker<type1,type2>::Initialize(DriveSituation& p_initialSituation,DriveSituation* p_testSituations, int p_testAmount);\
-    template bool DecisionMaker<type1,type2>::Decide(DriveSituation& p_driveSituation);\
+    template void DecisionMaker<type1,type2>::Initialize(tCarElt* p_initialCar, tSituation* p_initialSituation, BlackBoxData* p_testSituations, int p_testAmount);\
+    template bool DecisionMaker<type1,type2>::Decide(tCarElt* p_car, tSituation* p_situation, int p_tickCount);\
     template void DecisionMaker<type1,type2>::ChangeSettings(InterventionType p_type);\
     template void DecisionMaker<type1,type2>::SetDataCollectionSettings(tDataToStore p_dataSetting);
 
 #define TEMP_DECISIONMAKER DecisionMaker<SocketBlackBox,SDAConfig>
 
 /// @brief                     Initializes the decision maker
+/// @param  p_initialCar       The initial car
 /// @param  p_initialSituation The initial situation
 /// @param  p_testSituations   The test situations
 /// @param  p_testAmount       The amount of tests
 template <typename SocketBlackBox, typename SDAConfig>
-void DecisionMaker<SocketBlackBox, SDAConfig>::Initialize(DriveSituation& p_initialSituation,
-    DriveSituation* p_testSituations, int p_testAmount)
+void DecisionMaker<SocketBlackBox, SDAConfig>::Initialize(tCarElt* p_initialCar, 
+    tSituation* p_initialSituation, BlackBoxData* p_testSituations, int p_testAmount)
 {
-    m_blackBox.Initialize(p_initialSituation, p_testSituations, p_testAmount);
+    BlackBoxData initialData(p_initialCar, p_initialSituation, 0);
```

Comment:
This would only have a benefit in initial situatio0n, but shouldn't make a difference at all in the rest of the code. But because of that it is IMO easier to understand this (as it is the same as drivetick) than when I would use BlackBoxData